crates/nu-command/src/strings/encode_decode/encode.rs

```diff
@@ -15,7 +15,11 @@ impl Command for Encode {
     }
 
     fn usage(&self) -> &str {
-        "Encode an UTF-8 string into other kind of representations."
+        // Note: "Encode a UTF-8 string into other forms" is semantically incorrect because
+        // Nushell strings, as abstract values, have no user-facing encoding.
+        // (Remember that "encoding" exclusively means "how the characters are
+        // observably represented by bytes").
```

Comment:
Is this comment correct? A nushell `Value::String` holds a rust `String`. This is part of the documentation for rust Strings.

```
/// `String`s are always valid UTF-8. If you need a non-UTF-8 string, consider
/// [`OsString`]. It is similar, but without the UTF-8 constraint. Because UTF-8
/// is a variable width encoding, `String`s are typically smaller than an array of
/// the same `chars`:
```
Maybe I'm just not understanding the point you're trying to make?

Reply:
The point is whether their encoding is *user-facing* or *observable*. In most non-systems programming languages/scripting languages, strings' implementation encodings are not end-user observable because they are *abstract*. The whole point of the word "encoding" is that the user must treat the strings as *not* being any particular sequence of *codes*, but as the "pure characters" in and of themselves.

Reply:
I feel like this is a bit of a grey area. Strings tend to be important enough that even in higher-level languages, their encoding is well-known and guaranteed even when it's not directly observable (i.e. how .NET uses UTF-16 internally).

In Nushell's case, `into binary` does a raw conversion from `String` to a `Vec<u8>` so I think you could make the case that the encoding is observable in Nushell.

I don't have a well-formed opinion on this yet, but at first glance I think we'd be OK publicly defining Nushell strings as UTF-8.

Reply:
Our string indexing in `str substring` also follows Rust UTF-8 rules. So implicitly we have leaned on string always being UTF-8 encoded and using byte indices.

Reply:
`str substring` breaking on UTF-8 byte boundaries feels like a bug to me. As an end-user I would expect it to break on codepoints rather than bytes. For instance, `"🥇" | str substring '0,1'` should give `"🥇"` instead of just a useless character. For comparision, Javascript's new (2016+) string iterator correctly breaks strings on codepoints: `[..."🥇"]` produces `["🥇"]`.